internal/nullable/value.go

```diff
@@ -16,10 +16,17 @@ type Value[T any] struct {
 	Value T
 }
 
-// FromString[T any] represents a parser function. It is used to avoid
+func New[T any](v T, set bool) Value[T] {
+	return Value[T]{
+		Set:   set,
+		Value: v,
+	}
+}
+
```

Comment:
What do you think about below approach ? 

```suggestion
type Nullable interface {
	~string | ~int | ~uint
}

// Value type is used to represent difference betweeen an
// intentionally omitted value and default type value.
type Value[T Nullable] struct {
	Set   bool
	Value T
}

func New[T Nullable](v T) Value[T] {
	return NewNull[T].Set(v)
}

func NewNull[T Nullable]() Value[T] {
	return Value[T]{}
}

// Set sets the value and marks it as set.
func (v *Value[T]) Set(val T) {
	v.Set = true
	v.Value = val
}

```

Reply:
If `T` is `any` we could not determine `v == nil`
```go
func New[T](v T) Value[T] {
   if v == nil  { // x not possible 
   ...
}
```

If we have constraint like below, then it makes sure that pointer or interface can't be passed so `New[T Nullable]` function can set without checking for `nil`
```go
type Nullable interface {
	~string | ~int | ~uint
}
```

```
func New[T Nullable](v T) Value[T] {
	return NewNull[T].Set(v)
}
```

Reply:
I have added a union, but let's keep the factory method signature as is - it's simple enough and works for all cases. Setting the value is a nice idea, but we'd have to make it non-exported and use getters and setter, which is not Go idiomatic, and also does not bring anything to the table unless we make both `set` and `value` non-exported. Then, we can enforce that Set always makes `set` true, but I'm not sure that's even desired behavior. New(value, bool) should be enough and it's also the most flexible.